File: AdaBoost.java
# 🔋 AdaBoost in Java using Weka

This project demonstrates the implementation of the **AdaBoost** algorithm in **Java** using the Weka machine learning library.

## 🛠️ Requirements

- Java 8 or above
- Weka library (add `weka.jar` to your classpath)
- ARFF dataset file (e.g., `data.arff`)

## 📄 Description

We use `AdaBoostM1` with a simple base learner `DecisionStump` to classify data.

## 📦 How to Run

1. Place your dataset (e.g., `data.arff`) in the root folder.
2. Compile and run the code:

import weka.classifiers.meta.AdaBoostM1;
import weka.classifiers.trees.DecisionStump;
import weka.core.Instances;
import weka.core.converters.ConverterUtils.DataSource;

public class AdaBoostExample {
    public static void main(String[] args) throws Exception {
        // Load dataset
        DataSource source = new DataSource("data.arff");
        Instances data = source.getDataSet();
        if (data.classIndex() == -1)
            data.setClassIndex(data.numAttributes() - 1);

        // Setup AdaBoost
        AdaBoostM1 boost = new AdaBoostM1();
        boost.setClassifier(new DecisionStump());
        boost.setNumIterations(10);
        boost.buildClassifier(data);

        System.out.println(boost);
    }
}

```bash
javac -cp weka.jar AdaBoostExample.java
java -cp .:weka.jar AdaBoostExample

File: AdaBoost.cpp
### 📁 `cpp-boosting/README.md`

```markdown
# ⚙️ Boosting in C++ using XGBoost

This project implements **Gradient Boosting** in **C++** using the official **XGBoost C++ API**.

## 🛠️ Requirements

- C++11 or above
- XGBoost (compiled from source)

## 🔧 Build XGBoost

Follow official instructions to build XGBoost:
https://xgboost.readthedocs.io/en/stable/build.html

## 📄 Dataset

The training data must be in `libsvm` format (e.g., `train.svm`)

## 🚀 How to Run

1. Place your dataset `train.svm` in the root.
2. Compile the code:

```bash
g++ boost.cpp -o boost -I/path/to/xgboost/include -L/path/to/xgboost/lib -lxgboost
